Possible regression in type parameter nullability matching in 0.13.2

The following compiles successfully with 0.13.1 but fails with 0.13.2 (and 0.13.3):

```java
package com.example;

import org.jspecify.annotations.Nullable;
import org.jspecify.annotations.NullMarked;

import java.util.Map;
import java.util.function.Function;
import java.util.stream.Stream;

@NullMarked
public class Example {

	public Example and(Map<String, String> map) {
		return and(map.entrySet().stream(), Pair::fromMapEntry);
	}

	public <T> Example and(Stream<T> stream, Function<T, @Nullable Pair> mapper) {
		return this;
	}

	static class Pair {

		private final String name;

		private final @Nullable String value;

		private Pair(String name, @Nullable String value) {
			this.name = name;
			this.value = value;
		}

		static Pair fromMapEntry(Map.Entry<String, @Nullable String> entry) {
			return new Pair(entry.getKey(), entry.getValue());
		}

	}

}
```

The above is extracted from Spring Boot's [`TestPropertyValues`](https://github.com/spring-projects/spring-boot/blob/a71dd41fb864ecb46c319eb4f91fff6b99d05a8a/core/spring-boot-test/src/main/java/org/springframework/boot/test/util/TestPropertyValues.java). It fails to compile:

```
…/src/main/java/com/example/Example.java:14: error: [NullAway] parameter type of referenced method is Entry<String, @Nullable String>, but parameter in functional interface method has type Entry<String, String>, which has mismatched type parameter nullability
                return and(map.entrySet().stream(), Pair::fromMapEntry);
                                                    ^
    (see http://t.uber.com/nullaway )
1 error
```

Passing `Entry<String, String>` into a method that accepts `Entry<String, @Nullable String>` looks OK to me, hence reporting it as a possible regression.

--- a/nullaway/src/test/java/com/uber/nullaway/jspecify/GenericMethodLambdaOrMethodRefArgTests.java
+++ b/nullaway/src/test/java/com/uber/nullaway/jspecify/GenericMethodLambdaOrMethodRefArgTests.java
@@ -964,6 +964,48 @@ public class GenericMethodLambdaOrMethodRefArgTests extends NullAwayTestsBase {
   }
 
   @Test
+  public void issue1528_methodRefParamTypeAllowsNonNullTypeArgForNullableTypeArg() {
+    makeHelperWithInferenceFailureWarning()
+        .addSourceLines(
+            "Test.java",
+            """
+            import org.jspecify.annotations.NullMarked;
+            import org.jspecify.annotations.Nullable;
+            import java.util.Map;
+            import java.util.function.Function;
+            import java.util.stream.Stream;
+            @NullMarked
+            class Test {
+              <T> Test and(Stream<T> stream, Function<T, @Nullable Pair> mapper) {
+                return this;
+              }
+              static class Pair {
+                private final String name;
+                private final @Nullable String value;
+                private Pair(String name, @Nullable String value) {
+                  this.name = name;
+                  this.value = value;
+                }
+                static Pair fromMapEntry(Map.Entry<String, @Nullable String> entry) {
+                  return new Pair(entry.getKey(), entry.getValue());
+                }
+                static Pair fromMapEntryWildcard(Map.Entry<String, ? extends @Nullable String> entry) {
+                  return new Pair(entry.getKey(), entry.getValue());
+                }
+              }
+              Test and(Map<String, String> map) {
+                // BUG: Diagnostic contains: parameter type of referenced method is Entry<String, @Nullable String>
+                return and(map.entrySet().stream(), Pair::fromMapEntry);
+              }
+              Test andWildcard(Map<String, String> map) {
+                return and(map.entrySet().stream(), Pair::fromMapEntryWildcard);
+              }
+            }
+            """)
+        .doTest();
+  }
+
+  @Test
   public void genericMethodLambdaArgWildCard() {
     makeHelperWithInferenceFailureWarning()
         .addSourceLines(

--- a/nullaway/src/test/java/com/uber/nullaway/jspecify/GenericMethodLambdaOrMethodRefArgTests.java
+++ b/nullaway/src/test/java/com/uber/nullaway/jspecify/GenericMethodLambdaOrMethodRefArgTests.java
@@ -964,7 +964,7 @@ public class GenericMethodLambdaOrMethodRefArgTests extends NullAwayTestsBase {
   }
 
   @Test
-  public void issue1528_methodRefParamTypeAllowsNonNullTypeArgForNullableTypeArg() {
+  public void issue1528() {
     makeHelperWithInferenceFailureWarning()
         .addSourceLines(
             "Test.java",

